File: Data Structures & Algorithms/subarray-sum-equals-k/submission-2.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} k
     * @return {number}
     */
    subarraySum(nums, k) {
        let count = 0;
        for(let i = 0; i < nums.length; i += 1) {
            let sum = 0;
            for(let j = i; j < nums.length; j += 1) {
                sum += nums[j];
                if(sum === k) {
                    count += 1;
                }
            }
        }
        return count;
    }
}

File: Data Structures & Algorithms/top-k-elements-in-list/submission-2.py
import heapq
from typing import List

class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        
        freqMap = {}
        for i in range(len(nums)):
            curr_num = nums[i]
            if curr_num not in freqMap:
                freqMap[curr_num] = 1
            else:
                freqMap[curr_num] +=1

        max_pq = []
        for key, val in freqMap.items():
            heapq.heappush_max(max_pq, (val, key))

        result = []
        for _ in range(k):
            (count, num) = heapq.heappop_max(max_pq)
            result.append(num)
        
        return result